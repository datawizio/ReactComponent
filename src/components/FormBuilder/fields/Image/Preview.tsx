import React from "react";

import { DeleteOutlined, ExclamationCircleOutlined } from "@ant-design/icons";
import { Modal } from "antd";
import { useTranslation } from "react-i18next";

export interface PreviewProps {
  value: string;
  onDelete: () => void;
}

export const Preview: React.FC<PreviewProps> = ({ value, onDelete }) => {
  const { t } = useTranslation();

  const handleClick = (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
    e.stopPropagation();
    Modal.confirm({
      title: t("DELETE_CONFIRM_PHOTO"),
      icon: <ExclamationCircleOutlined />,
      okText: t("YES"),
      cancelText: t("CANCEL"),
      onOk() {
        onDelete();
      },
      onCancel() {}
    });
  };

  return (
    <div className="ant-upload-list ant-upload-list-picture-card">
      <div className="ant-upload-list-picture-card-container">
        <span>
          <div
            className="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
            onClick={handleClick}
          >
            <div className="ant-upload-list-item-info">
              <span>
                <img
                  src={value}
                  alt="Preview"
                  className="ant-upload-list-item-image"
                />
              </span>
            </div>
            <span className="ant-upload-list-item-actions">
              <DeleteOutlined />
            </span>
          </div>
        </span>
      </div>
    </div>
  );
};
