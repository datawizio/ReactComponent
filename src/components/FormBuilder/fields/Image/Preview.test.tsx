import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import { Preview } from "./Preview";

const mockProps = {
  name: "Preview name",
  placeholder: "placeholder",
  value: "value",
  onChange: jest.fn()
};

const setUp = (props?) => shallow(<Preview {...props} />);

describe("FormBuilder Preview component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Preview rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("Preview call  modal", () => {
    const mockEvent = {
      stopPropagation: jest.fn()
    };
    const modalBtn = component.find(".ant-upload-list-item").first();
    modalBtn.simulate("click", mockEvent);
    expect(component).toMatchSnapshot();
  });
});
