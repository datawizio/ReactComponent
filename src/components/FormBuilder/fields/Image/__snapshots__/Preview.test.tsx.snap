// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBuilder Preview component Preview call  modal 1`] = `
<div
  className="ant-upload-list ant-upload-list-picture-card"
>
  <div
    className="ant-upload-list-picture-card-container"
  >
    <span>
      <div
        className="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
        onClick={[Function]}
      >
        <div
          className="ant-upload-list-item-info"
        >
          <span>
            <img
              alt="Preview"
              className="ant-upload-list-item-image"
              src="value"
            />
          </span>
        </div>
        <span
          className="ant-upload-list-item-actions"
        >
          <ForwardRef(DeleteOutlined) />
        </span>
      </div>
    </span>
  </div>
</div>
`;

exports[`FormBuilder Preview component Preview rendered correctly 1`] = `
<div
  className="ant-upload-list ant-upload-list-picture-card"
>
  <div
    className="ant-upload-list-picture-card-container"
  >
    <span>
      <div
        className="ant-upload-list-item ant-upload-list-item-done ant-upload-list-item-list-type-picture-card"
        onClick={[Function]}
      >
        <div
          className="ant-upload-list-item-info"
        >
          <span>
            <img
              alt="Preview"
              className="ant-upload-list-item-image"
              src="value"
            />
          </span>
        </div>
        <span
          className="ant-upload-list-item-actions"
        >
          <ForwardRef(DeleteOutlined) />
        </span>
      </div>
    </span>
  </div>
</div>
`;
